### ISSUE TITLE
should we drop the vertical line in mcmc_areas() and mcmc_areas_ridges()?

### ISSUE THREAD

**Issue by jgabry** [stan-dev/bayesplot#195]
I'm starting to think that maybe we shouldn't be hardcoding the vertical line at zero at all. Users don't have the option to remove and it would be super easy for them to add a vertical afterwards if they want it. We could consider suggesting users add `+ vline_0()` (or `+ geom_vline(xintercept=0)` directly) after making the plot if they want the vertical line. 

I'm not saying we should definitely change this, just thinking out load. Of course if someone wants the plot without the line they do already have the option of getting the data from `mcmc_*_data()` and making it themselves.

**Comment by jgabry** [comment]
I suppose another option is to have an argument to toggle it.

**Comment by avehtari** [comment]
Makes sense either way

**Comment by utkarshpawade** [comment]
hi @jgabry @avehtari, the vertical line is still hardcoded in `mcmc_intervals()`, `mcmc_areas()`, and `mcmc_areas_ridges()`, I'd lean towards toggle suggestion over dropping the line outright, since removing it silently would change every existing plot on first upgrade. Something like a `vline_at_zero` argument defaulting to TRUE would let users opt out without breaking existing code or docs. If the line is dropped later, the argument can be deprecated cleanly. Can i raise a small pr if it sounds good ?

**Comment by avehtari** [comment]
Instead of argument, maybe a global package option would be better?

**Comment by utkarshpawade** [comment]
yes, that matches the existing bayesplot.base_size / bayesplot.base_family pattern nicely. One option is a hybrid: a function argument that defaults to the global option, e.g. vline_at_zero = getOption("bayesplot.mcmc_vline_at_zero", TRUE). That way users can flip it globally once, override per plot when needed, and it's still documented in ?mcmc_areas. Happy to proceed with whichever you prefer: global or hybrid.

**Comment by utkarshpawade** [comment]
yes, that matches the existing bayesplot.base_size / bayesplot.base_family pattern, which actually uses a hybrid (function arg defaulting to getOption(...)). Pure global option works too; the hybrid just has the advantage of showing up in ?mcmc_areas and allowing per-plot overrides. Happy to proceed with whichever you prefer. One small note: the same hardcoded line is also in mcmc_intervals(), so I'd assume that's in scope too unless you'd rather keep it as-is there.

**Comment by avehtari** [comment]
We are in progress of planning global options, and this should wait until we have guidelines for how to name global options (we have some inconsistencies in existing global options)
